javascript/modules/client/src/client.ts

```diff
@@ -483,6 +495,7 @@ export class Client extends ClientCore implements IClient {
 // @ts-ignore  TODO: Remove this comment
 function unwrapCreateDaoParams(
   params: ICreateParams,
+  metadataIpfsUri: string
 ): [DAOFactory.DAOConfigStruct, DAOFactory.VoteConfigStruct, string, string] {
   // TODO: Serialize plugin params into a buffer
```

Comment:
The goal of this function should be to receive developer friendly parameters and rearrange them in terms that are smart contract friendly. The IPFS payload should be part of the developer submitted payload. Does `ICreateParams` include the metadata object? At some point the JSON string should be uploaded to IPFS, but this is outside of this function's scope.
Let's carefully think about that when we implement the final functionality

Reply:
Yes, `ICreateParams` include the metadata as an Object, the idea is to upload this object to IPFS and here we introduce the received `cid`

Reply:
Ok, but then... is this function called for handling the response? or for receiving the user data and eventually transforming it for the blockchain? If it's the second case, then I guess the function should be called `wrapXXX` instead? 

I'm not clear at all what's the purpose of the function at this point, stringifying the received object should not be this function's job... at least someone else serializing and uploading to IPFS before, then wrapping the params here

Reply:
Yes, this is exactly the idea behind this, you already uploaded the metadata to ipfs and already have a `cid`, now you have this and a bunch of other input parameters that you want to use to call a function on the SCs, so you call this function, and you receive an array with the exact format that you will need to call the function in the `core-contracts-ethers` package.

About the `unwrap` naming, is the same naming that we've been using in the rest of the code for functions that have the exact same purpose but in other context like `unwrapWithdrawParams`, `unwrapErc20InitParams`, `unwrapErc20InitParams`...

Reply:
So, regargind the IPFS data, I would suggest that this function receives the CiD as opposed to the full object then. Because it could get serialized in different ways (with spaces, without spaces, etc) and create a different CiD than the intended one